sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/PhysicsConstraintGizmo.cs

```diff
@@ -0,0 +1,564 @@
+            var shapeWorldMatrix = shapeLocalMatrix * rigidbody.Entity.Transform.WorldMatrix;
+            // we don't want to scale the pivot model, or the pivot translation
+            ResetScale(ref shapeWorldMatrix);
+
+            // we want the pivot marker to receive scaling from the gizmo system
+            var targetScale = GizmoScalingEntity.Transform.Scale;
+
+            var pivotPosition = Matrix.Scaling(targetScale) * Matrix.Translation(pivot) * shapeWorldMatrix;
+
+            pivotMarker.Entity.Transform.UseTRS = false;
+            pivotMarker.Entity.Transform.LocalMatrix = pivotPosition;
+
+            // and ensure the model is enabled
+            pivotMarker.Enable(true);
+        }
+
+        #region Cache accessors
+        private static GeometricPrimitive GetSphere(Graphics.GraphicsDevice device)
+        {
+            if (!SphereCache.TryGetValue(device, out var sphere))
+            {
+                sphere = GeometricPrimitive.Sphere.New(device, CenterSphereRadius, Tessellation);
+                SphereCache.Add(device, sphere);
+            }
+
+            return sphere;
+        }
+
+        private static GeometricPrimitive GetCone(Graphics.GraphicsDevice device)
+        {
+            if (!ConeCache.TryGetValue(device, out var cone))
+            {
+                cone = GeometricPrimitive.Cone.New(device, AxisConeRadius, AxisConeHeight, Tessellation);
+                ConeCache.Add(device, cone);
+            }
+
+            return cone;
+        }
+
+        private static GeometricPrimitive GetCylinder(Graphics.GraphicsDevice device)
+        {
+            if (!CylinderCache.TryGetValue(device, out var cylinder))
+            {
+                cylinder = GeometricPrimitive.Cylinder.New(device, CylinderLength, CylinderRadius, Tessellation);
+                CylinderCache.Add(device, cylinder);
+            }
+
+            return cylinder;
+        }
+
+        private static GeometricPrimitive GetLimitDisc(Graphics.GraphicsDevice device, float angle)
+        {
+            if (!LimitDiscCache.TryGetValue(device, out var cache))
+            {
+                cache = new Dictionary<float, GeometricPrimitive>();
+                LimitDiscCache.Add(device, cache);
+            }
+
+            if (!cache.TryGetValue(angle, out var disc))
+            {
+                disc = GeometricPrimitive.Disc.New(device, LimitDiscRadius, angle, Tessellation);
+                cache.Add(angle, disc);
+            }
+
+            return disc;
+        }
+
+        private static Material GetMaterial(Graphics.GraphicsDevice device, Color color)
+        {
+            if (!MaterialCache.TryGetValue(device, out var cache))
+            {
+                cache = new Dictionary<Color, Material>();
+                MaterialCache.Add(device, cache);
+            }
+
+            if (!cache.TryGetValue(color, out var material))
+            {
+                material = GizmoUniformColorMaterial.Create(device, color);
```

Comment:
the transformation gizmos use `GizmoEmissiveColorMaterial`, I think this one should too? (in order to have the same style)

Reply:
Good point, otherwise it might disappear without lighting.

Reply:
Looks kind of harsh to me and doesn't seem to support transparency ( I could use the uniform color for just the limit disc )
I mean, not sure what the preference here is from people, visually - for me the only important thing is that it's functional.
![image](https://user-images.githubusercontent.com/10709060/129726015-b929f36d-3b05-41fe-844a-450ab7fb6d59.png)

Reply:
@manio143 sure, please use whatever you feel is better UX wise.

Reply:
I added transparency support to the emissive material (copied from the uniform one) and used slightly lower intensity.
![image](https://user-images.githubusercontent.com/10709060/130322961-a396e56a-083d-4306-85a2-9e9211ccaf59.png)